backend/app/project_handler.go

```diff
@@ -0,0 +1,130 @@
+package app
+
+import (
+	"errors"
+
+	"github.com/codescalersinternships/Flyspray/models"
+	"github.com/gin-gonic/gin"
+	"github.com/rs/zerolog/log"
+	"gorm.io/gorm"
+)
+
+type createProjectInput struct {
+	Name string `json:"name" binding:"required"`
+}
+
+type updateProjectInput struct {
+	Name string `json:"name" binding:"required"`
+}
+
+func (a *App) createProject(ctx *gin.Context) (interface{}, Response) {
+	var input createProjectInput
+
+	if err := ctx.BindJSON(&input); err != nil {
+		log.Error().Err(err).Send()
+		return nil, BadRequest(errors.New("failed to read project data"))
+	}
+
+	// check if project name exists before
+	if _, err := a.client.GetProjectByName(input.Name); err != gorm.ErrRecordNotFound {
+		log.Error().Err(err).Send()
+		return nil, BadRequest(errors.New("project name must be unique"))
+	}
+
+	// TODO: get user id from authorization middleware and assign it to OwnerId
+	newProject := models.Project{Name: input.Name, OwnerId: 10007} // 10007 is just a random number
+	newProject, err := a.client.CreateProject(newProject)
+
+	if err != nil {
+		log.Error().Err(err).Send()
+		return nil, InternalServerError(errors.New("failed to create project"))
+	}
+
+	return ResponseMsg{
+		Message: "project is created successfully",
+		Data:    newProject,
+	}, Created()
+}
+
+func (a *App) updateProject(ctx *gin.Context) (interface{}, Response) {
+	// TODO: get user id from authorization middleware and check if user has access to update the project
+	id := ctx.Param("id")
+	var input updateProjectInput
+
+	if err := ctx.BindJSON(&input); err != nil {
+		log.Error().Err(err).Send()
+		return nil, BadRequest(errors.New("failed to read project data"))
+	}
+
+	// check if project name exists before
+	if _, err := a.client.GetProjectByName(input.Name); err != gorm.ErrRecordNotFound {
+		log.Error().Err(err).Send()
+		return nil, BadRequest(errors.New("project name must be unique"))
```

Comment:
same here, also the error could be nil or it could be another error (like can't connect to db) so not sure this is the best way to handle it

Reply:
you can just update the project without getting it first and check affected rows to determine if it is not found